### DISCUSSION TITLE
How to install the yarn workspace to all 'packages/*' at one time?

### DISCUSSION THREAD

**Discussion by ghost** [yarnpkg/berry#2789] [Q&A] [answered]
## bad
`yarn workspace package-0 add react`
react install to package-0

`yarn workspace package-1 add react`
react install to package-1

## good
`yarn workspace * add react`
`yarn workspace all add react`
`yarn workspaces add react`

----

Use lerna + yarn
lerna does not support deleting dependent packages
yarn does not support installation and deletion for all packages
This is where the problem lies

Or is there any compromise to deal with this problem?

**Comment by merceyz** [comment, marked as answer]
You can use `workspaces foreach` for that https://yarnpkg.com/cli/workspaces/foreach
```sh
yarn workspaces foreach add react
```
